### DISCUSSION TITLE
Money is an item

### DISCUSSION THREAD

**Discussion by AbsDataStruct** [ChestShop-authors/ChestShop-3#491] [Ideas]
"Money is an item" is an enhancement feature that would enable the use of a specific item (such emerald) as money rather than the money balance of economy system.

Minecraft is a blocks and items game, I like the way we can see all what we have in Chest or Inventory. Money balance is not as visual as a stock of wood in a chest. Children and rich players would love to see how much money they really have in stack of money in stacks of chests.
When the "money is an item" enhancement is activated, a specified item such emerald would become the coin used to sell and buy items for all shops, storing stack of money in Inventory or Chest.

The way to buy and sell with a ChestShop would be then only a simple exchange of money item such emerald with the purchased item. The Shop's Chest itself would continue to contain only the item to sell or buy, but allowing the owner to put many other chests under the top shop chest itself to store the owner's money used for transaction.

When a player BUY items, he would received items from the ShopChest as usual but the amount of money to pay would be taken from his Inventory as stack of the money item and put in one of multiple chests placed under the ShopChest.
When a player SELL items, he would put his items into the ShopChest as usual but the amount of money to earn would be taken from the multiple chests placed under the ShopChest and put into his Inventory as stack of the money item.

It is easy to develop and I'm trying to set a development environment in order to write the code myself ...
Any help is welcome, it's my first plugin development

**Comment by Phoenix616** [comment]
Just fyi.: This is already kind of possible with an economy plugin which uses items as currency like [Gringotts](https://www.spigotmc.org/resources/gringotts.42071/).

**Reply by AbsDataStruct** [reply]
Oh ! thanks so much ! I have searched that for so long time !!!!!
